### PR TITLE
service-load-balancing: Support service-linked role for appautoscaling

### PR DESCRIPTION
**Pull request by nishigori**
Currently, terraform-provider-aws is not support creating srevice-linked role yet.

Hope to attached service-liked role if user specified.

like: https://github.com/terraform-providers/terraform-provider-aws/issues/2750#issuecomment-354632543

## Refs:

- https://docs.aws.amazon.com/IAM/latest/UserGuide/using-service-linked-roles.html

### CODE REVIEW

**Comment by nishigori** [comment]
Still I'm testing to my aws resources now ...

**Comment by tamakiii** [comment]
:lgtm:

File: service_load_balancing/autoscaling.tf [hunk 1]
```diff
@@ -45,7 +45,7 @@ resource "aws_appautoscaling_target" "main" {
   max_capacity       = "${var.autoscale_max_capacity}"
   min_capacity       = "${var.autoscale_min_capacity}"
   resource_id        = "service/${var.cluster_name}/${aws_ecs_service.main.name}"
-  role_arn           = "${aws_iam_role.service_autoscale.arn}"
+  role_arn           = "${ var.autoscale_iam_role_arn != "" ? var.autoscale_iam_role_arn : aws_iam_role.service_autoscale.arn }"
```
Comment (nishigori) [inline service_load_balancing/autoscaling.tf]: Oops, conditional operator cannot declare resource...

https://github.com/hashicorp/terraform/issues/11566

**Comment by nishigori** [comment]
> https://github.com/hashicorp/terraform/issues/11566

This PR has BC BREAKS...

**Comment by nishigori** [comment]
Released [v0.2.0](https://github.com/voyagegroup/tf_aws_ecs/releases/tag/v0.2.0)

**Pull request merged**